v-next/core/src/internal/parameters.ts

```diff
@@ -54,3 +58,99 @@ const parameterTypeValidators: Record<
   [ParameterType.FLOAT]: (value): value is number => typeof value === "number",
   [ParameterType.FILE]: (value): value is string => typeof value === "string",
 };
+
+/**
+ * Parses a parameter value from a string to the corresponding type.
+ */
+// TODO: this code is duplicated in v-next/hardhat/src/internal/cli/main.ts
```

Comment:
I think exposing these things from core makes sense. We already expose functions like this. e.g. `resolvePluginList`.

Reply:
Agreed. I'll do it on a separate PR.

Reply:
Issue created: https://github.com/NomicFoundation/hardhat/issues/5431